### PR TITLE
Testing improvement

### PR DESCRIPTION
**Pull request by Akado2009**
- Massive testing refactoring
- Incorporating meta_state testing to improve testing

### CODE REVIEW

File: tests/mtk_tests.rs [hunk 1]
```diff
@@ -1,514 +1,192 @@
-use codec::Encode;
 use gear_lib::multitoken::io::*;
-use gstd::{ActorId, String};
-use gtest::{Program, System};
-use multitoken_io::*;
+use gstd::String;
+use gtest::System;
+
```
Comment (fluiderson) [inline tests/mtk_tests.rs]: You can remove these imports...

File: tests/utils.rs [hunk 2]
```diff
@@ -0,0 +1,270 @@
+use codec::Encode;
+use gear_lib::multitoken::{io::*, state::*};
+use gstd::{ActorId, String};
+use gtest::{Program, System};
```
Comment (fluiderson) [inline tests/utils.rs]: ...and add `pub` to those! I think it's shorter and more convenient.
```suggestion
pub use gear_lib::multitoken::{io::*, state::*};
use gstd::{ActorId, String};
pub use gtest::{Program, System};
```

File: tests/mtk_tests.rs [hunk 3]
```diff
@@ -1,514 +1,192 @@
...
-            ids: vec![TOKEN_ID],
-            amounts: vec![TOKEN_AMOUNT],
-        }
-        .encode()
-    )));
-}
-
 #[test]
-fn mint() {
+fn mint__() {
```
Comment (fluiderson) [inline tests/mtk_tests.rs]: What are these underscores at the end for?

File: tests/mtk_tests.rs [hunk 4]
```diff
@@ -1,514 +1,190 @@
-use codec::Encode;
 use gear_lib::multitoken::io::*;
```
Comment (fluiderson) [inline tests/mtk_tests.rs]: This can also be removed.
```suggestion
```

**Review by fluiderson — APPROVED** [review]

**Review by shamilsan — APPROVED** [review]
Great 👍

**Review by LouiseMedova — APPROVED** [review]

**Pull request merged**